src/test/java/teammates/ui/webapi/UpdateFeedbackSessionActionTest.java

```diff
@@ -94,6 +99,240 @@ protected void testExecute() {
+
+        FeedbackSessionUpdateRequest updateRequest = getTypicalFeedbackSessionUpdateRequest();
+        Map<String, Long> newStudentDeadlines = updateRequest.getStudentDeadlines()
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().toEpochMilli()));
+        newStudentDeadlines.put("student1InCourse1@gmail.tmt", 1809126000000L);
+        updateRequest.setStudentDeadlines(newStudentDeadlines);
+
+        UpdateFeedbackSessionAction a = getAction(updateRequest, param);
+        JsonResult r = getJsonResult(a);
+        FeedbackSessionData response = (FeedbackSessionData) r.getOutput();
+
+        Map<String, Long> expectedStudentDeadlines = new HashMap<>();
+        expectedStudentDeadlines.put("student1InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student3InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student4InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student5InCourse1@gmail.tmt", 1809126000000L);
+        assertEquals(expectedStudentDeadlines, response.getStudentDeadlines());
+
+        ______TS("update deadline extension for student");
+
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+        newStudentDeadlines = updateRequest.getStudentDeadlines()
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().toEpochMilli()));
+        newStudentDeadlines.put("student1InCourse1@gmail.tmt", 1809212400000L);
+        updateRequest.setStudentDeadlines(newStudentDeadlines);
+
+        a = getAction(updateRequest, param);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        expectedStudentDeadlines = new HashMap<>();
+        expectedStudentDeadlines.put("student1InCourse1@gmail.tmt", 1809212400000L);
+        expectedStudentDeadlines.put("student3InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student4InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student5InCourse1@gmail.tmt", 1809126000000L);
+        assertEquals(expectedStudentDeadlines, response.getStudentDeadlines());
+
+        ______TS("delete deadline extension for student");
+
+        // the typical update request does not contain the course 1 student 1's email
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+
+        a = getAction(updateRequest, param);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        expectedStudentDeadlines = new HashMap<>();
+        expectedStudentDeadlines.put("student3InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student4InCourse1@gmail.tmt", 1809126000000L);
+        expectedStudentDeadlines.put("student5InCourse1@gmail.tmt", 1809126000000L);
```

Comment:
Same for the rest, consider adding a comment to show which deadlines have been deleted.